add minimax-m1 doc (#59)
vllm-project/vllm-project.github.io: _posts/2025-06-26-minimax-m1.md

* **Right:** Theoretical inference FLOPs scaling with token length. Compared to DeepSeek R1, MiniMax-M1 uses only 25% of the FLOPs when generating sequences of 100k tokens.

## Deploying MiniMax-M1 with vLLM

We recommend deploying MiniMax-M1 using **vLLM** for optimal performance. Our tests demonstrate the following key benefits:

- Outstanding throughput
- Efficient and intelligent memory management
- Robust support for batched requests
- Deeply optimized backend performance

### Model Download

You can download the models from Hugging Face:

```bash
# Install the Hugging Face Hub CLI
pip install -U huggingface-hub

# Download the MiniMax-M1-40k model
huggingface-cli download MiniMaxAI/MiniMax-M1-40k
# For the 80k version, uncomment the following line:
# huggingface-cli download MiniMaxAI/MiniMax-M1-80k
```

### Deployment

Below is a quick guide to deploying MiniMax-M1 with vLLM and Docker. Each step is annotated for clarity:

```bash
# Set environment variables
IMAGE=vllm/vllm-openai:latest 
MODEL_DIR=<model storage path> 
NAME=MiniMaxImage 

# Docker run configuration

> DOCKER_RUN_CMD="--network=host --privileged --ipc=host --ulimit memlock=-1 --shm-size=2gb --rm --gpus all --ulimit stack=67108864"

Reviewer: nit: `--shm-size=2gb` would be unnecessary when we have `--ipc=host` .

Author: Update

- Outstanding throughput
- Efficient and intelligent memory management
- Robust support for batched requests
- Deeply optimized backend performance

### Model Download

You can download the models from Hugging Face:

```bash
# Install the Hugging Face Hub CLI
pip install -U huggingface-hub

# Download the MiniMax-M1-40k model
huggingface-cli download MiniMaxAI/MiniMax-M1-40k
# For the 80k version, uncomment the following line:
# huggingface-cli download MiniMaxAI/MiniMax-M1-80k
```

### Deployment

Below is a quick guide to deploying MiniMax-M1 with vLLM and Docker: 

```bash
# Set environment variables
IMAGE=vllm/vllm-openai:latest 
MODEL_DIR=<model storage path> 
NAME=MiniMaxImage 

# Docker run configuration
DOCKER_RUN_CMD="--network=host --privileged --ipc=host --ulimit memlock=-1 --rm --gpus all --ulimit stack=67108864"

# Start the container
sudo docker run -it \
    -v $MODEL_DIR:$MODEL_DIR \    
    --name $NAME \                
    $DOCKER_RUN_CMD \          
    $IMAGE /bin/bash    

# Launch MiniMax-M1 Service
export SAFETENSORS_FAST_GPU=1
export VLLM_USE_V1=0

> python3 -m vllm.entrypoints.openai.api_server \

Reviewer 2: nit: I think we can simplify it a bit!
```suggestion
vllm serve \
```

vLLM incorporates a wide range of CUDA kernel optimizations, including integrations with FlashAttention, FlashInfer, and support for quantization formats such as GPTQ, AWQ, INT4, INT8, and FP8.

These enhancements further boost the low-level computation efficiency of MiniMax-M1 inference. Quantization reduces memory and compute overhead with minimal accuracy loss, while FlashAttention accelerates the attention computation itself—resulting in significantly faster inference in real-world applications.

### Lightning Attention in vLLM

As a cutting-edge attention mechanism, Lightning Attention is implemented in vLLM via Triton, leveraging its flexibility and high-performance computing features. A Triton-based execution framework fully supports Lightning Attention's core computation logic, enabling seamless integration and deployment within the vLLM ecosystem.

## Future Work

Looking ahead, further optimizations for hybrid architecture support are actively being explored within the vLLM community. Notably, the development of a hybrid allocator is expected to enable even more efficient memory management tailored to the unique requirements of models like MiniMax-M1.

> In addition, full support for [vLLM v1](https://minimax-m1.vllm-blog-source.pages.dev/2025/01/27/v1-alpha-release) is planned, with the hybrid model architecture expected to be migrated into the v1 framework. These advancements are anticipated to unlock further performance improvements and provide a more robust foundation for future developments.

Reviewer: ```suggestion
In addition, full support for [vLLM v1](https://blog.vllm.ai/2025/01/27/v1-alpha-release.html) is planned, with the hybrid model architecture expected to be migrated into the v1 framework. These advancements are anticipated to unlock further performance improvements and provide a more robust foundation for future developments.
```

## Conclusion

The hybrid architecture of MiniMax-M1 paves the way for the next generation of large language models, offering powerful capabilities in long-context reasoning and complex task inference. vLLM complements this with highly optimized memory handling, robust batch request management, and deeply tuned backend performance.

Together, MiniMax-M1 and vLLM form a strong foundation for efficient and scalable AI applications. As the ecosystem evolves, we anticipate this synergy will power more intelligent, responsive, and capable solutions across a wide range of use cases, including code generation, document analysis, and conversational AI.

## Acknowledgement

> We would like to express our sincere gratitude to the vLLM community for their invaluable support and collaboration. In particular, we thank [Tyler Michael Smith](https://github.com/tlrmchlsmth), [Simon Mo](https://github.com/simon-mo), [Cyrus Leung](https://github.com/DarkLight1337), [Roger Wang](https://github.com/ywang96), [Zifeng Mo](https://github.com/Isotr0py) and [Kaichao You](https://github.com/youkaichao) for their significant contributions. We also appreciate the efforts of the MiniMax engineering team, especially [Gangying Qing](https://github.com/ZZBoom), [Jun Qing](https://github.com/qscqesze), and [Jiaren Cai](https://github.com/sriting), whose dedication made this work possible.

Reviewer: @tlrmchlsmth @simon-mo @DarkLight1337 @ywang96 @Isotr0py can you please take a look and check if you have any suggestions?

the blog is planned to be released this week, thanks!

> We would like to express our sincere gratitude to the vLLM community for their invaluable support and collaboration. In particular, we thank [Tyler Michael Smith](https://github.com/tlrmchlsmth), [Simon Mo](https://github.com/simon-mo), [Cyrus Leung](https://github.com/DarkLight1337), [Roger Wang](https://github.com/ywang96), [Isotr0py](https://github.com/Isotr0py) and [Kaichao You](https://github.com/youkaichao) for their significant contributions. We also appreciate the efforts of the MiniMax engineering team, especially [Gangying Qing](https://github.com/ZZBoom), [Jun Qing](https://github.com/qscqesze), and [Jiaren Cai](https://github.com/sriting), whose dedication made this work possible.

Reviewer: ```suggestion
We would like to express our sincere gratitude to the vLLM community for their invaluable support and collaboration. In particular, we thank [Tyler Michael Smith](https://github.com/tlrmchlsmth), [Simon Mo](https://github.com/simon-mo), [Cyrus Leung](https://github.com/DarkLight1337), [Roger Wang](https://github.com/ywang96), [Zifeng Mo](https://github.com/Isotr0py) and [Kaichao You](https://github.com/youkaichao) for their significant contributions. We also appreciate the efforts of the MiniMax engineering team, especially [Gangying Qing](https://github.com/ZZBoom), [Jun Qing](https://github.com/qscqesze), and [Jiaren Cai](https://github.com/sriting), whose dedication made this work possible.
```
